figures/mapping-histology-labels.Rmd

```diff
@@ -0,0 +1,237 @@
+---
+title: "Mapping histology labels for plots"
+output:   
+  html_notebook: 
+    toc: true
+    toc_float: true
+author: Candace Savonen for ALSF - CCDL
+date: 2020
+---
+
+# Purpose: 
+
+The histology label variables included in `pbta-histologies.tsv` from data releases, while perhaps specific and accurate, are not useful for summarizing the biospecimens for visuals. 
+
+The goal of this notebook is to use the currently existing `broad_histology` groups from `pbta-histologies.tsv`, to form 10-15 "high level histology" group labels that can used for plotting purposes.
+
+## The output table
+
+The output of this notebook is a TSV file: `palettes/histology_label_color_table.tsv` that contains...
+  
+**Copied from `pbta-histologies.tsv`**:    
+- `Kids_First_Biospecimen_ID` (from `pbta-histologies.tsv`)  
+- All the original histology label variables (`broad_histology`, `short_histology`, etc.)  
+  
+**Created in this notebook**:  
+- `summary_group` - the high-level histology labels that should be used for plotting   
+- `hex_codes` the direct colors that should be used for plotting  
+
+With this info, `histology_label_color_table.tsv` should be used by all plots and figures that need summarizing on a high-level. 
+
+# How `summary_group` is made:
+
+Here's how `broad-histology` is [recoded into the high-level groupings of `summary_group`](#make_new_summary_group).
+
+1) All biospecimens in `broad_histology` groups with less than the number of `small_groups_cutoff` will be now grouped together as `Other` in `summary_group`. 
+
+2) "Neuronal and mixed neuronal-glial tumor" and "Diffuse astrocytic and oligodendroglial tumor" were labeled as LGAT in previous releases so we will group these into `Low-grade astrocytic tumor` in `summary_group`. 
+
+3) So its clear why they are `NA`s, we will label all `Normal` samples (non-tumor samples) as `Normal` in `summary_group`. 
+
+4) Anything not in the above categories gets its `broad_histology` label carried over. 
+
+# Usage
+
+This notebook can be run via the command line from the top directory of the 
+repository as follows:
+
+```
+Rscript -e "rmarkdown::render('figures/mapping-histology-labels.Rmd', 
+                              clean = TRUE)"
+```
+
+## Set Up
+
+```{r}
+# Magrittr pipe
+`%>%` <- dplyr::`%>%`
+```
+
+For groups that are smaller than this cutoff, they will be put into an `Other` group in `summary_group`. 
+
+```{r}
+small_groups_cutoff <- 12
+```
+
+### Directories and Files
+
+```{r}
+# Path to input directory
+input_dir <- file.path("..", "data")
+output_dir <- "palettes"
+```
+
+# Read in metadata 
+
+Which variables are we keeping for this table? 
+
+```{r}
+histology_variables <- 
```

Comment:
For ease of downstream use, I'm wondering if we should drop everything that's not `Kids_First_Biospecimen_ID` `summary_group` and `hex_codes`, because downstream we'll need to drop everything. OR do we save a separate file just like `histology_color_palette.tsv` but that has each biospecimen's `summary_group` and `hex_codes`.